### ISSUE TITLE
Extensions have to specify patterns in the config file, can't hard-code in the source

### ISSUE THREAD

**Issue by cubranic** [radiocosmology/alpenhorn#70]
Example config:
```
database:
    url: mysql://root@127.0.0.1/alpenhorn_db

logging:
    level: INFO

extensions:
    - frb_data_types
```

frb_data_types.py:
```py
from alpenhorn import generic as ge

# Create handlers for the acquisition and file types
class FrbEventInfo(ge.GenericAcqInfo):
    _acq_type = 'frb'
    _file_types = [
    ]
    # this is using globre syntax, with `{}` escaping into RE
    patterns = [r'20[12][0-9]/[0-9][0-9]/[0-9][0-9]/astro_{\\d+}']

def register_extension():
    return {
        'acq_types': [FrbEventInfo],
        'file_types': [],
    }
```

Traceback:
```
  File "/Users/davor/projects/archiving/a2/alpenhorn/client.py", line 54, in init
    extensions.register_type_extensions()
  File "/Users/davor/projects/archiving/a2/alpenhorn/extensions.py", line 113, in register_type_extensions
    _register_acq_extensions(ext_dict['acq_types'])
  File "/Users/davor/projects/archiving/a2/alpenhorn/extensions.py", line 131, in _register_acq_extensions
    acq_type.set_config(conf)
  File "/Users/davor/projects/archiving/a2/alpenhorn/generic.py", line 43, in set_config
    cls.patterns = configdict['patterns']
KeyError: 'patterns'
```

**Comment by jrs65** [comment]
I think it would be fine for me to change the underlying code in `GenericAcqInfo` to work around this. 

However, if you're hard coding your own patterns you might as well just base off `AcqInfoBase` directly. The extra functionality provided be `GenericAcqInfo` is small (there's probably only 20 lines of actual code) and you're not using most of it.

**Comment by cubranic** [comment]
Fair enough, I'll probably do just that. But it is nice when experimenting to just subclass Generic and focus on just the relevant bits. So you almost convinced me not to do it, even though I already have the changes ready to go. What do you prefer?

**Comment by ketiltrout** [comment]
Fixed by #149 (by pushing all the import detect logic to the extensions).
